service/backend/pchain/indexer/parser.go

```diff
@@ -0,0 +1,318 @@
+package indexer
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"time"
+
+	"github.com/ava-labs/avalanchego/codec"
+	"github.com/ava-labs/avalanchego/genesis"
+	"github.com/ava-labs/avalanchego/ids"
+	"github.com/ava-labs/avalanchego/snow"
+	"github.com/ava-labs/avalanchego/utils/constants"
+	"github.com/ava-labs/avalanchego/utils/hashing"
+	"github.com/ava-labs/avalanchego/utils/wrappers"
+	"github.com/ava-labs/avalanchego/vms/platformvm/blocks"
+	pGenesis "github.com/ava-labs/avalanchego/vms/platformvm/genesis"
+	"github.com/ava-labs/avalanchego/vms/platformvm/txs"
+	"github.com/ava-labs/avalanchego/vms/proposervm/block"
+
+	"github.com/ava-labs/avalanche-rosetta/client"
+	"github.com/ava-labs/avalanche-rosetta/mapper"
+	"github.com/ava-labs/avalanche-rosetta/service/backend/common"
+)
+
+var errParserUninitialized = errors.New("uninitialized parser")
+
+const genesisTimestamp = 1599696000
+
+type Parser interface {
+	Initialize(ctx context.Context) (*ParsedGenesisBlock, error)
```

Comment:
q: It feels weird having an `Initialize` method returning the genesis. Why not splitting the initiazation from the retrieval of genesis block, which may be carried out with e.g.  `ParseBlockAtIndex`?